File: Python/Math/1351_count_negative_numbers_in_a_sorted_matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        # Function to check if the list contains negatives
        def isListSumEqual(n: List[int]) -> bool:
            nSum = 0
            absSum = 0
            
            for i in range(len(n)):
                nSum += n[i]
                absSum += abs(n[i])
            
            return nSum == absSum
        
        count = 0
        
        for i in range(len(grid)):
            # If the list contains negatives
            if (not isListSumEqual(grid[i])):
                for j in range(len(grid[i])):
                    # Count negatives
                    if (grid[i][j] < 0):
                        count += 1
        
        return count

File: Swift/Math/1351_count_negative_numbers_in_a_sorted_matrix.swift
class Solution {
    func countNegatives(_ grid: [[Int]]) -> Int {
        // Use the Idea of Binary Sort
        // Function to find the first occurence of negative value
        func findNegativeIndex(n: [Int]) -> Int {
            if (n[0] < 0) {
                return 0
            }
            
            var left = 0
            var right = n.count / 2
            
            while (right < n.count) {
                // Negative value is in left side
                if (n[right - 1] < 0) {
                    right -= (right - left) / 2
                }
                // Negative value is n[right]
                else if (n[right] < 0) {
                    return right
                }
                // Negative value is in right side
                else {
                    left = right
                    right += (n.count - right) / 2
                }
            }
            
            return right
        }
        
        var count = 0
        for i in (0..<grid.count) {
            // If there's negative value in grid[i]
            if (grid[i].min()! < 0) {
                count += grid[i].count - findNegativeIndex(n: grid[i])
            }
        }
        return count
    }
}
